wgpu-core/src/device/ray_tracing.rs

```diff
@@ -0,0 +1,376 @@
+                        vertex_stride: 0,
+                        indices,
+                        transform: None,
+                        flags: desc.flags,
+                    });
+                }
+                unsafe {
+                    self.raw().get_acceleration_structure_build_sizes(
+                        &hal::GetAccelerationStructureBuildSizesDescriptor {
+                            entries: &hal::AccelerationStructureEntries::Triangles(entries),
+                            flags: blas_desc.flags,
+                        },
+                    )
+                }
+            }
+        };
+
+        let raw = unsafe {
+            self.raw()
+                .create_acceleration_structure(&hal::AccelerationStructureDescriptor {
+                    label: blas_desc.label.as_deref(),
+                    size: size_info.acceleration_structure_size,
+                    format: hal::AccelerationStructureFormat::BottomLevel,
+                })
+        }
+        .map_err(DeviceError::from_hal)?;
+
+        let handle = unsafe {
+            self.raw()
+                .get_acceleration_structure_device_address(raw.as_ref())
+        };
+
+        Ok(Arc::new(resource::Blas {
+            raw: ManuallyDrop::new(raw),
+            device: self.clone(),
+            size_info,
+            sizes,
+            flags: blas_desc.flags,
+            update_mode: blas_desc.update_mode,
+            handle,
+            label: blas_desc.label.to_string(),
+            built_index: RwLock::new(rank::BLAS_BUILT_INDEX, None),
+            tracking_data: TrackingData::new(self.tracker_indices.blas_s.clone()),
+        }))
+    }
+
+    fn create_tlas(
+        self: &Arc<Self>,
+        desc: &resource::TlasDescriptor,
+    ) -> Result<Arc<resource::Tlas>, CreateTlasError> {
+        let size_info = unsafe {
+            self.raw().get_acceleration_structure_build_sizes(
+                &hal::GetAccelerationStructureBuildSizesDescriptor {
+                    entries: &hal::AccelerationStructureEntries::Instances(
+                        hal::AccelerationStructureInstances {
+                            buffer: None,
+                            offset: 0,
+                            count: desc.max_instances,
+                        },
+                    ),
+                    flags: desc.flags,
+                },
+            )
+        };
+
+        let raw = unsafe {
+            self.raw()
+                .create_acceleration_structure(&hal::AccelerationStructureDescriptor {
+                    label: desc.label.as_deref(),
+                    size: size_info.acceleration_structure_size,
+                    format: hal::AccelerationStructureFormat::TopLevel,
+                })
+        }
+        .map_err(DeviceError::from_hal)?;
+
+        let instance_buffer_size = get_raw_tlas_instance_size(self.backend())
+            * std::cmp::max(desc.max_instances, 1) as usize;
```

Comment:
nit: I think calling `desc.max_instances.max(1)` would be a little more expected